apis/v1beta1/config.go

```diff
@@ -133,6 +139,53 @@ type Config struct {
 	Service    Service    `json:"service" yaml:"service"`
 }
 
+// getPortsForComponentTypes gets the ports for the given ComponentType(s).
+// NOTE FOR REVIEWERS: we could also do this by introducing something in components that is aware of which retriever
```

Comment:
Seeking opinions here

Reply:
I guess this depends on where we want to handle finding the right component config. At this level, we *know* it exists, as we're iterating over the map. The retriever would have to check and potentially return an error. Is there any other place we need this logic in? If it's just here, I'd keep it as is.